frontend/assets/javascripts/src/modules/analytics/setup.js

```diff
@@ -34,30 +34,43 @@ define([
         !cookie.getCookie('ANALYTICS_OFF_KEY')
     );
 
-    function setupAnalytics() {
-        ga.init();
+    function reportTagLoadFail(tracker, allPurposesAgreed) {
+        console.log(`Either there's insufficient consent for ${tracker.vendorName}, or the user has ` +
+            `turned that vendor off in the CMP (${tracker.cmpVendorId}). ` +
+            `The user has ${allPurposesAgreed ? '' : 'not'} agreed to all purposes.`);
     }
 
-    function setupThirdParties() {
-        facebook.init();
-        uet.init();
-        remarketing.init();
-    }
-
-    function init() {
-        campaignCode.init();
+    function loadTrackers() {
+        const trackers = [ga, facebook, uet, remarketing];
+        const vendorIds = trackers.map(tracker => tracker.cmpVendorId);
 
-        if (analyticsEnabled) {
-            setupAnalytics();
-        }
+        Promise.allSettled([
+            cmp.checkCCPA(),
+            cmp.getConsentForVendors(vendorIds),
+            cmp.checkAllTCFv2PurposesAreOptedIn(),
+        ]).then(results => {
+            const [ccpaConsent, vendorConsents, allPurposesAgreed] = results.map(promise => promise.value);
 
-        thirdPartyTracking.thirdPartyTrackingEnabled().then(thirdPartyTrackingEnabled => {
-            if (analyticsEnabled && thirdPartyTrackingEnabled && !guardian.isDev) {
-                setupThirdParties();
+            if (ccpaConsent) {
+                trackers.forEach(tracker => tracker.init());
+                campaignCode.init();
+            } else {
+                trackers.forEach(tracker => {
+                    console.log('Attempting to load ' + tracker.vendorName);
+                    vendorConsents[tracker.cmpVendorId] ?
```

Comment:
In CCPA mode without consent, all trackers will always be `false`. I'm assuming this allows have a log of which ones are prevented from running.